### PR TITLE
Added ENABLE_BACKTRACE_SYMBOLS guard for dladdr usage with backtraces.

### PR DESCRIPTION
**Pull request by mcfadden8**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -37,6 +37,7 @@ option(ENABLE_EXAMPLES "Build Umpire examples" On)
 option(ENABLE_LOGGING "Build Umpire with Logging enabled" On)
 option(ENABLE_SLIC "Build Umpire with SLIC logging" Off)
 option(ENABLE_BACKTRACE "Build Umpire with allocation backtrace enabled" Off)
+option(ENABLE_DLSYMS "Build Umpire with symbol support" Off)
```
Comment (davidbeckingsale) [inline CMakeLists.txt:40]: Do you think that this should/could depend on `ENABLE_BACKTRACE`? I think that it should be on by default when backtrace is, which maintains the current behavior.

Also, can you come up with a better name? Maybe `ENABLE_BACKTRACE_SYMBOLS`?

Comment (mcfadden8) [reply CMakeLists.txt]: Good idea.  This is done.

**Comment by mcfadden8** [comment]
@davidbeckingsale,  this should now be ready.  You may see example outputs in the jira ticket located here: https://rzlc.llnl.gov/jira/browse/UM-631

File: docs/sphinx/advanced_configuration.rst [hunk 2]
```diff
@@ -68,6 +69,9 @@ These arguments are explained in more detail below:
 * ``ENABLE_BACKTRACE``
   This option enables collection of backtrace information for each allocation.
 
+* ``ENABLE_BACKTRACE_SYMBOLS``
+  This option enables symbol information to be provided with backtraces.
```
Comment (davidbeckingsale) [inline docs/sphinx/advanced_configuration.rst:73]: Please say this requires `-ldl`

Comment (mcfadden8) [reply docs/sphinx/advanced_configuration.rst]: Done.

**Comment by davidbeckingsale** [comment]
Windows passed too :)

**Comment by davidbeckingsale** [comment]
LGTM.

**Pull request merged**